bin/annotate_test_failures

```diff
@@ -16,14 +18,17 @@ slack_channel = nil
 slack_webhook = ENV['SLACK_WEBHOOK'] # default value inferred from env var if not provided explicitly
 args = OptionParser.new do |opts|
   opts.banner = <<~HELP
-    Usage: annotate_test_failures [junit-report-file-path] [--slack CHANNEL] [--slack-webhook URL] 
+    Usage: annotate_test_failures [junit-report-file-path] [--slack CHANNEL] [--slack-webhook URL]
 
     Annotates the Buildkite build with a summary of failed and flaky tests based on a JUnit report file (defaults to using `build/results/report.junit`).
     Optionally also posts the same info to a Slack channel.
   HELP
-  opts.on('--slack CHANNEL_NAME', 'The name of the Slack channel to post the failure summary to') { |v| slack_channel = '#' + v.delete_prefix('#') }
+  opts.on('--slack CHANNEL_NAME', 'The name of the Slack channel to post the failure summary to') { |v| slack_channel = "##{v.delete_prefix('#')}" }
```

Comment:
Nitpick: I find three usages of `#` here a bit confusing. Not sure if this suggestion would make it better or worse 😄 
```suggestion
  opts.on('--slack CHANNEL_NAME', 'The name of the Slack channel to post the failure summary to') { |v| slack_channel = v.delete_prefix('#').prepend('#') }
```